### PR TITLE
mon: paxos small optimization on store_state

### PR DESCRIPTION
**Pull request by songbaisen**
mon: paxos small optimization on store_state

Signed-off-by: song baisen song.baisen@zte.com.cn

### CODE REVIEW

File: src/mon/Paxos.cc [hunk 1]
```diff
@@ -358,6 +358,7 @@ bool Paxos::store_state(MMonPaxos *m)
     dout(10) << "store_state ignoring all values, they start at " << start->first
 	     << " > last_committed+1" << dendl;
     start = m->values.end();
+    return changed;
```
Comment (tchaikov) [inline src/mon/Paxos.cc]: if we want to early return here, maybe we can remove the line of `start = m->values.end();` also.

**Comment by songbaisen** [comment]
@tchaikov  Yes,Thank you for view.I have done it.

**Comment by jecluis** [comment]
Also, personally, I think returning just 'false' would be a bit more readable than returning 'changed' (which is set to false at this point).

**Comment by songbaisen** [comment]
@jecluis  Yes,Done it.It make clear here.

**Comment by athanatos** [comment]
@jecluis It's ok to skip remove_legacy_versions() in that case?

**Comment by athanatos** [comment]
This has passed tests (sjust@teuthology:/a/samuelj-2016-05-16_20:02:21-rados-wip-sam-testing-distro-basic-smithi).  Once someone reviews it, it can merge.

@tchaikov @jecluis Is this ok?

**Comment by tchaikov** [comment]
lgtm.

**Pull request merged**